decidim-core/app/jobs/decidim/event_publisher_job.rb

```diff
@@ -4,21 +4,26 @@ module Decidim
   class EventPublisherJob < ApplicationJob
     queue_as :events
 
-    attr_reader :resource
+    attr_reader :resource, :data
 
     def perform(event_name, data)
       @resource = data[:resource]
+      @data = data
 
       return unless data[:force_send] || notifiable?
 
-      EmailNotificationGeneratorJob.perform_later(
-        event_name,
-        data[:event_class],
-        data[:resource],
-        data[:followers],
-        data[:affected_users],
-        data[:extra]
-      )
+      if event_type.include?(:email)
```

Comment:
Why do you add this change? Are we planning sending in app notifications that do not have a email to back up ?

Reply:
I've added this just to respect the api explained in the `BaseEvent`, in which the "type" of the event should define how the event is going to be treated. In here, if sent by email or web notification (other types could be added like web-push, but that is automatic for the type notification).
`SimpleEvent` has the two concerns `EmailEvent` and `NotifcationEvent` which means it will be sent by both methods, but ti does not stop developers to define their custom events and define which deliver system they should prefer.

In this PR, the event `CoauthorAcceptedInviteEvent` does not inherith from `SimpleEvent` but directly from `BaseEvent` and includes only the concen `NotificationEvent`, that's because I consider this notification useful to see in your area of notification (where you perform the action of accepting a person to a group) but worthy of bothering via email.

I think this behavior makes Decidim more versatile.